server/webapp/WEB-INF/rails/webpack/views/components/expand_collapsible/index.js.msx

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const m      = require('mithril');
+const Stream = require('mithril/stream');
+const styles = require('./index.scss');
+
+module.exports = {
+  oninit(vnode) {
+    const self    = this;
+    self.expanded = Stream(false);
+
+    self.toggle = () => {
+      self.expanded(!self.expanded());
+    };
+
+    self.showState = () => {
+      return self.expanded() ? 'show' : 'hide';
+    }
+  },
+
+  view(vnode) {
+    const self = this;
+
+    return (
+      <div class={`${styles.collapse} ${self.expanded() ? styles.expanded : styles.collapsed}`}>
```

Comment:
We've added a module called added [classNames](https://github.com/JedWatson/classnames). Some of these conditional styles can be written like
```javascript
const classes = classNames({
  styles.collapse,
  styles.expanded: self.expanded()
  styles.collapsed: !self.expanded()
})
```
or something similar